pkg/distributor/distributor.go

```diff
@@ -500,9 +595,28 @@ func (d *Distributor) sendSamples(ctx context.Context, ingester ring.IngesterDes
 	}
 	_, err = c.Push(ctx, &req)
 
-	ingesterAppends.WithLabelValues(ingester.Addr).Inc()
+	ingesterAppends.WithLabelValues(ingester.Addr, typeSamples).Inc()
+	if err != nil {
+		ingesterAppendFailures.WithLabelValues(ingester.Addr, typeSamples).Inc()
+	}
+	return err
+}
+
+func (d *Distributor) sendMetadata(ctx context.Context, ingester ring.IngesterDesc, metadata []client.PreallocMetricMetadata) error {
+	h, err := d.ingesterPool.GetClientFor(ingester.Addr)
+	if err != nil {
+		return err
+	}
+	c := h.(ingester_client.IngesterClient)
+
+	req := client.WriteRequest{
+		Metadata: metadata,
+	}
+	_, err = c.Push(ctx, &req)
+
+	ingesterAppends.WithLabelValues(ingester.Addr, typeMetadata).Inc()
 	if err != nil {
-		ingesterAppendFailures.WithLabelValues(ingester.Addr).Inc()
+		ingesterAppendFailures.WithLabelValues(ingester.Addr, typeMetadata).Inc()
```

Comment:
For this metric, I have added a new label `type` to identify what is metadata and what is samples, given they are all ingester appends.